viewer/src/public/migration/Cognite3DModel.ts

```diff
@@ -233,6 +233,42 @@ export class Cognite3DModel extends THREE.Object3D implements CogniteModelBase {
     return this.determineTreeIndices(treeIndex, true);
   }
 
+  /**
+   * Determines the tree index range of a subtree of an ancestor of the provided
+   * node defined by a tree index.
+   *
+   * @param treeIndex     Tree index of node to find ancestor tree index range for.
+   * @param ancestorLevel What ancestor level to find. 0 is the node itself,
+   *                      1 means parent, 2 means grandparent etc. If the node doesn't
+   *                      have as many ancestors, the root of the model is returned. This can
+   *                      be determined by checking that the range returned includes 0.
+   * @returns             Tree index range of the subtree spanned by the ancestor at level ancestorLevel.
+   */
+  async getAncestorTreeIndices(treeIndex: number, ancestorLevel: number): Promise<NumericRange> {
+    // This might fail if treeIndex is invalid, error checking below is just to be safe
+    // and should not really happen.
+    const nodeId = await this.mapTreeIndexToNodeId(treeIndex);
+
+    const ancestors = await this.client.revisions3D.list3DNodeAncestors(this.modelId, this.revisionId, nodeId, {
+      limit: 1000
+    });
+    const node = ancestors.items.find(x => x.treeIndex === treeIndex);
+    if (node === undefined) {
```

Comment:
Maybe better to use `assert`. Feel like assert is great for those "should never happen" things.